assets/src/components/v2/triptych/evergreen_content.tsx

```diff
@@ -0,0 +1,51 @@
+import React, { ComponentType } from "react";
+
+import EvergreenContent from "../evergreen_content";
+import useIsOnScreen from "Hooks/v2/use_is_on_screen";
+import { imagePath } from "Util/util";
+import { TRIPTYCH_VERSION } from "./version";
+import { usePlayerName } from "Hooks/outfront";
+import { TriptychPane, getTriptychPane } from "Util/outfront";
+
+const isInViewport = (thisComponentsPane: TriptychPane) => {
+  const triptychPane = getTriptychPane();
+  if (triptychPane == null) {
+    // Viewport is not limited/shifted, we're showing all 3 in a browser tab.
+    return true;
+  } else {
+    return thisComponentsPane == triptychPane;
+  }
+};
+
+/**
+ * Differences from standard evergreen content component:
+ * - allows showing identifiers (version, player name) over content
+ * - does not render content outside the viewport at all, besides identifiers, to save on work
+ */
+const TriptychEvergreenContent: ComponentType<{
+  asset_url: string;
+  show_identifiers: boolean;
+  triptychPane: TriptychPane;
+}> = ({
+  asset_url: assetUrl,
+  show_identifiers: showIdentifiers,
+  triptychPane,
+}) => {
+  const dupReadyUrl = imagePath(assetUrl);
+  const isPlaying = useIsOnScreen();
+  const playerName = usePlayerName();
+  let identifiers = `${TRIPTYCH_VERSION} ${playerName ? playerName : ""}`;
+
+  return (
+    <>
+      {isInViewport(triptychPane) && (
```

Comment:
Could this viewport logic actually live above this in `triptych_three_pane`? We have access to what we need there too so could just make this a layout change instead of a widget change.

Reply:
Oh duh, yeah I think it could go there. I would just need to make sure that the middle and left screens still draw in the right positions if the left is not rendered.

Will make this change and then reassign you.

Reply:
So much simpler now, thanks for pointing that out.